tests/integration_tests/instances.py

```diff
@@ -25,6 +26,26 @@ def _get_tmp_path():
     return '/var/tmp/{}.tmp'.format(tmp_filename)
 
 
+class CloudInitSource(Enum):
+    """Represents the cloud-init image source setting as a defined value.
+
+    Values here represent all possible values for CLOUD_INIT_SOURCE in
+    tests/integration_tests/integration_settings.py. See that file for an
+    explanation of these values. If the value set there can't be parsed into
+    one of these values, an exception will be raised
+    """
+    NONE = 1
+    IN_PLACE = 2
+    PROPOSED = 3
+    PPA = 4
+    DEB_PACKAGE = 5
+
+    def installs_new_version(self):
```

Comment:
Something's not working for me here and the tests are skipped. The method extension on this Enum subclass is falling apart somehow:
```bash
> /home/csmith/src/cloud-init/tests/integration_tests/test_upgrade.py(38)test_upgrade()
-> if not source.installs_new_version():
(Pdb) str(source)
'CloudInitSource.NONE'
(Pdb) dir(source)
['__class__', '__doc__', '__module__', 'installs_new_version', 'name', 'value']
(Pdb) source.installs_new_version()
*** SyntaxError: invalid syntax
(Pdb) type(source)
<enum 'CloudInitSource'>
(Pdb) source

*** SyntaxError: unexpected EOF while parsing
```
ation_settings.CLOUD_INIT_SOURCE == 'PROPOSED':
￼        clie

Reply:
I believe we resolved this on Friday?

Reply:
Yes Resolved. I hadn't specified CLOUD_INIT_SOURCE=PROPOSED